### PR TITLE
cpu/stm32f7: add riotboot requirements

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR adds riotboot for stm32f7. Since F7 works with sectors for riotboot to work each slot must start at the beginning of a sector. For stm32f7 the smallest sector is of length 16kB or 32kb depending on CPU_MODEL, therefore that must be the length of the bootloader and the same amount must be taken from the second half of the flash to optimize flash usage.

This PR is part of 3, since just adding riotboot doesn't make much sense if it can't perform ota. Although it doesn't depend on them but it is highly related to:

* Support flashpage: https://github.com/RIOT-OS/RIOT/pull/11681
* Support flashpage raw for flashwrite: https://github.com/RIOT-OS/RIOT/pull/11705

Also pulling https://github.com/RIOT-OS/RIOT/pull/11808 can ease testing significantly.

### Testing procedure

It has been tested on the supported boards: `nucleo-f746zg` and `nucleo-f767zi`

1.- Test riotboot:

`BOARD=nucleo-f746zg FEATURES_REQUIRED+=riotboot make -C tests/xtimer_usleep riotboot/flash-extended-slot0 term`

`BOARD=nucleo-f746zg FEATURES_REQUIRED+=riotboot make -C tests/xtimer_usleep riotboot/flash-slot1 term`

In both cases it should run without issues.

2.- Since the sector division is important when updating the slots by writing to flash directly it would be ideal to rebase on top off flashpage: https://github.com/RIOT-OS/RIOT/pull/11681 and https://github.com/RIOT-OS/RIOT/pull/11705, as well as https://github.com/RIOT-OS/RIOT/pull/11808

and run `tests/riotboot_flashwrite`

The following patch can be applied to the test Makefile

`make -C tests/riotboot_flashwrite/ BOARD=nucleo-f746zg flash test`

### Issues/PRs references

* Support flashpage: https://github.com/RIOT-OS/RIOT/pull/11681
* Support flashpage raw for flashwrite: https://github.com/RIOT-OS/RIOT/pull/11705

* Also pulling https://github.com/RIOT-OS/RIOT/pull/11808 can ease testing significantly.

### CODE REVIEW

File: cpu/stm32f7/Makefile.include [hunk 1]
```diff
@@ -1,5 +1,28 @@
 export CPU_ARCH = cortex-m7
 export CPU_FAM  = stm32f7
 
+# STM32F7 uses sectors instead of pages, where the minimum sector length is 16KB or
+# 32kB (the first sector), depending on the CPU_MODEL. Therefore RIOTBOOT_LEN must
+# be 16KB or 32kB to cover a whole sector.
+ifneq (,$(filter stm32f722ze ,$(CPU_MODEL)))
```
Comment (aabadie) [inline cpu/stm32f7/Makefile.include]: This is a specific case for stm32f722ze so I would either add riotboot for nucleo-f722ze or remove this part. I prefer the former so I'll test on the board and report the result in this PR.

Comment (fjmolinas) [reply cpu/stm32f7/Makefile.include]: It is not really a specific for `stm32f722ze`, a lot of `stm32f7` have 16kB sector. It's just that we only support 4 `f7`, and the other 3 have 32kB sized sectors.

All `stm32f7x2xx` hace 16kB and all `stm32f76xxx` that are smaller than 2Mb. It really depends on the `stm32f7` sub-family.

Comment (aabadie) [reply cpu/stm32f7/Makefile.include]: Sorry, but if you add this if, you add a case for stm32f722ze.

Anyway, I tested this PR on nucleo-f722ze and it works. So you can add the riotboot feature for this board as well.

Comment (fjmolinas) [reply cpu/stm32f7/Makefile.include]: Ok, should I leave the if as it is? The other option would be to add defines for the `stm32f7` sub-families.

Comment (aabadie) [reply cpu/stm32f7/Makefile.include]: >  should I leave the if as it is?

Yes, but add then just add riotboot feature to nucleo-f722ze and I'll ACK this PR.

Comment (fjmolinas) [reply cpu/stm32f7/Makefile.include]: In that case do you have stm32769i-disco so we add it for all stm32f7 board in one go?

Comment (aabadie) [reply cpu/stm32f7/Makefile.include]: Unfortunately, we don't have this one.

Comment (fjmolinas) [reply cpu/stm32f7/Makefile.include]: Ok, I didn't include it then.

**Comment by fjmolinas** [comment]
@aabadie thanks for the review!

**Pull request merged**